### PR TITLE
OMS Storage Analytucs V2

### PR DESCRIPTION
**Pull request by Volkanco**

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@Volkanco can you check the CI failures?

**Comment by Volkanco** [comment]
@bmoore-msft  I have tested  this deployment many times  with powershell using the  parameter files and it deploys just fine

**Comment by Volkanco** [comment]
I had the same CI problems in my initial release , it was always failing  without any specific errors . At that time I have been told that CI has some limitations and sometimes cant test it properly

**Comment by krnese** [comment]
Tested manually & verified.

File: oms-azure-storage-analytics-solution/README.md [hunk 1]
```diff
@@ -1,22 +1,24 @@
-# Azure Storage Analytics (Preview)
+# Azure Storage Analytics 
 
 [![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Foms-azure-storage-analytics-solution%2Fazuredeploy.json) 
 <a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Foms-azure-storage-analytics-solution%2Fazuredeploy.json" target="_blank">
     <img src="http://armviz.io/visualizebutton.png"/>
 </a>
 
->[AZURE.NOTE]This is preliminary documentation for Azure Storage Analytics, a management solution you can deploy into OMS that will provide insights and analytics of your storage accounts. The solution is currently in preview. 
+>[AZURE.NOTE]This is preliminary documentation for Azure Storage Analytics, a management solution you can deploy into OMS that will provide insights, analytics and auditing  for your storage accounts. This  Cummunity Resource Manager  template is not supported under any Microsoft support program or service and is made available AS IS without warranty of any kind!
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done...

File: oms-azure-storage-analytics-solution/README.md [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
 
  Microsoft Azure Storage is the cloud storage solution for modern applications that rely on durability, availability, and scalability to meet the needs of their customers. Azure storage provides  Blob storage, Table storage, Queue storage, and File storage services.
-Azure Storage Analytics Solution collects and visualizes inventory information, capacity metrics, transaction metrics and storage errors for storage accounts in an Azure subscription. Users can easily create additional rules to monitor storage resources. This solution leverages Azure Automation, the Log Analytics Ingestion API, together with Log Analytics views to present data about all your storage accounts  in a single  workspace. 
+Azure Storage Analytics Solution collects and visualizes inventory information, capacity metrics, transaction metrics,storage errors and Audit Logs  for storage accounts in an Azure subscription. Users can easily create additional rules to monitor storage resources. This solution leverages Azure Automation, the Log Analytics Ingestion API, together with Log Analytics views to present data about all your storage accounts  in a single  workspace.
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done

File: oms-azure-storage-analytics-solution/README.md [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
+![alt text](images/sasolution.png "Overview") ![alt text](images/logsolution.png "Overview")
 
 ## Pre-reqs
 
 - **Automation Account with SPN**
 
-Before you deploy this template, you must create an Automation Account in the Azure portal with the default settings so that the SPN account will be created. If you have an existing OMS Log Analytics Workspace you would like to use for this solution, it is important that the Automation account is created into the **same Resource Group where the OMS Log Analytics Workspace is located**.
+Before you deploy this template, you must create an Automation Account in the Azure portal with the default settings so that the SPN account will be created. If you have an existing OMS Log Analytics Workspace you would like to use for this solution, it is important that the Automation account is created into the **same Resource Group where the OMS Log Analytics Workspace is located**. 
+
+If you want to collect data from multiple subscription Automation Account SPN needs to have required permission on all subscriptions. See Multiple Subscription support section for details.
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done.

File: oms-azure-storage-analytics-solution/README.md [hunk 4]
```diff
@@ -52,6 +54,10 @@ Ensure that the parameters reflects your setup so that you are deploying this in
 
 #### Parameters
 
+* Deplyment Name Suffix 
+
+Specify a unique name to be used as suffix to deployment name. This name will be used to create a guid  needed for the automation jobs. You need to provide a unique suffix for the resource group every time you deploy the solution  .
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done

File: oms-azure-storage-analytics-solution/README.md [hunk 5]
```diff
@@ -68,17 +78,24 @@ Specify the name of the automation account you created earlier
...
 Select the Azure Region where your automation account is located
 
+* OMS Workspace Type 
+
+Specify if Log Analytics Workspace and Automation account is linked 
 
-You should also change the values for the *Ingest Scheduler Guid* and *Ingest Cleanup Guid*. You can generate your own using PowerShell with the following cmdlet:
+* Linked Workspace SKU
 
+Specify the SKu to be used in linked workspace. Only used if OMS Workspace Type is set to Linked
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: replaced all references to SKU

File: oms-azure-storage-analytics-solution/README.md [hunk 6]
```diff
@@ -68,17 +78,24 @@ Specify the name of the automation account you created earlier
...
+Select if you want to deploy additional Azure Storage Audit Log solution components
 
-![alt text](images/template.png "template")
+* Collection From All Subscriptions 
 
-The ingestion will start 5-10 minutes post deployment.
+Select if you want to enable data collection from all subscriptions where Automation SPN  has access to 
+
+
+The ingestion will start 5-10 minutes post deployment. IF you have created a new log analytics workspace  it might take up tp 30 mins for all fields to populate.
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done

File: oms-azure-storage-analytics-solution/README.md [hunk 7]
```diff
@@ -135,6 +152,39 @@ The views for Azure Storage Analytics  will give you an overview of all the stor
 
 Solution also includes 14 preconfigured alerts  which can be used to notify for when certain thresholds are reached.  You can view the alerts and modify thresholds  from Alerts under workspace settings.
 
+### Azure Storage Audit Logs
+
+The views for Azure Storage Analytics  will give you an overview of all the storage accounts  within your Azure Subscription.
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/README.md]: done

File: oms-azure-storage-analytics-solution/azuredeploy.json [hunk 8]
```diff
@@ -1,2502 +1,250 @@
...
+    "parameters": {
+        "deploymentNameSuffix": {
+            "type": "string",
+            "metadata": {
+                "description": "Provide an unique deployment names for each deployment  subsequent deployments"
+            }
+        },
+        "omsLogAnalyticsWorkspaceName": {
+            "type": "string",
+            "defaultValue": "",
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/azuredeploy.json]: removed both empty default values

File: oms-azure-storage-analytics-solution/metadata.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "itemDisplayName": "OMS - Azure Storage Analytics Solution",
-  "description": "Enables Azure Storage Analytics Solution in OMS. Solution collects Azure storage account inventory and storage usage metrics into OMS workspace.",
+  "description": "Enables Azure Storage Analytics and Logs Solution in OMS. Solution collects Azure storage account inventory, storage usage metrics and audit logs  into OMS workspace.",
   "summary": "This template deploys  Azure Storage Analytics solution with OMS workspace and Automation account.",
   "githubUsername": "volkanco",
-  "dateUpdated": "2017-2-15"
+  "dateUpdated": "2017-8-2"
```
[the remark this answers is not in the document]
Comment (Volkanco) [reply oms-azure-storage-analytics-solution/metadata.json]: all changes are done...

**Pull request merged**